addons/binding/org.openhab.binding.tankerkoenig/src/main/java/org/openhab/binding/tankerkoenig/internal/data/TankerkoenigService.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.tankerkoenig.internal.data;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+
+import org.apache.commons.io.IOUtils;
+import org.openhab.binding.tankerkoenig.internal.config.TankerkoenigListResult;
+import org.openhab.binding.tankerkoenig.internal.serializer.CustomTankerkoenigListResultDeserializer;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.Version;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/***
+ * Serivce class requesting data from tankerkoenig api and providing result objects
+ *
+ * @author Dennis Dollinger
+ *
+ */
+public class TankerkoenigService {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private Gson gson;
+
+    public TankerkoenigListResult getTankstellenListData(String apikey, String locationIDs) {
+        TankerkoenigListResult result = this.getTankerkoenigListResult(apikey, locationIDs);
+        return result;
+    }
+
+    private String getResponseString(String apikey, String locationIDs) throws IOException {
+
+        String urlbase = "https://creativecommons.tankerkoenig.de/json/prices.php?";
+        String urlcomplete = urlbase + "ids=" + locationIDs + "&apikey=" + apikey;
+        try {
+            String userAgent = "openHAB, Tankerkoenig-Binding Version ";
+            Version version = FrameworkUtil.getBundle(this.getClass()).getVersion();
+            userAgent = userAgent + version.toString();
+            URL url = new URL(urlcomplete);
+            URLConnection connection = url.openConnection();
+            connection.setRequestProperty("User-Agent", userAgent);
+            String response = IOUtils.toString(connection.getInputStream());
+            return response;
+        } catch (MalformedURLException e) {
+            return null;
+        }
+    }
+
+    private TankerkoenigListResult getTankerkoenigListResult(String apikey, String locationIDs) {
+        String jsonData = "";
+        try {
+            jsonData = getResponseString(apikey, locationIDs);
+        } catch (IOException e) {
+            logger.error("Error in getTankerkoenigListResult: {}", e.toString());
```

Comment:
1) I won't assume that you can execute line 71 in a meaningful way if we got an error. Please add the whole body in the method including the declaration of jsonData. Maybe its best to create some instance with isOk false, to return in the error case. You could add a method to `TankerkoenigListResult`
public static TankerkoenigListResult emptyResult() {
    TankerkoenigListResult emptyResult = new TankerkoenigListResult();
    emptyResult.setOk(false);
    return emptyResult;
}
2) Pass the error either in of the following two ways:
```java
// pass the error as argument, shows both message and stacktrace
 logger.warn("Error in getTankerkoenigListResult: ", e);
```
```java
// pass the error message as argument, ignoring stacktrace
 logger.warn("Error in getTankerkoenigListResult: {}", e.getMessage());
```